tests/cases/compiler/bigintIndex.ts

```diff
@@ -0,0 +1,25 @@
+// @target: es2015
+// @experimentalBigInt: true
+
+interface BigIntIndex<E> {
+    [index: bigint]: E; // should error
+}
+
+const arr: number[] = [1, 2, 3];
+let num: number = arr[1];
+num = arr["1"];
+num = arr[1n]; // should error
+
+let key: keyof any; // should be type "string | number | symbol"
+key = 123;
+key = "abc";
+key = Symbol();
+key = 123n; // should error
+
+// Show correct usage of bigint index: explicitly convert to string
+const bigNum: bigint = 0n;
+const typedArray = new Uint8Array(3);
+typedArray[bigNum] = 0xAA; // should error
```

Comment:
Need another test for class property and methods with bigint as name (shouldn't be allowed)